servlet/src/main/java/org/togglz/servlet/activation/ClientIpActivationStrategy.java

```diff
@@ -39,29 +43,91 @@ public String getName()
    @Override
    public boolean isActive(FeatureState featureState, FeatureUser user)
    {
-
       HttpServletRequest request = HttpServletRequestHolder.get();
       if (request != null) {
 
-         String allowedIpsParam = featureState.getParameter(PARAM_IPS);
-         List<String> allowsIps = Strings.splitAndTrim(allowedIpsParam, "[\\s,]+");
-
-         // TODO: This should support a simple form of subnet matching
-         return allowsIps.contains(request.getRemoteAddr());
-
+         List<String> parts = Strings.splitAndTrim(featureState.getParameter(PARAM_IPS), "[\\s,]+");
+
+         for (String part : parts) {
+            String remoteAddr = request.getRemoteAddr();
+            if (part.equals(remoteAddr)) { // shortcut
+               return true;
+            }
+            
+            try {
+               if (part.contains("/")) {
+                  CIDRUtils cidrUtil = new CIDRUtils(part);
+                  if (cidrUtil.isInRange(remoteAddr)) {
+                     return true;
+                  }
+               } else if (InetAddress.getByName(remoteAddr).equals(InetAddress.getByName(part))) {
```

Comment:
I think `InetAddress.getByName(remoteAddr)` should be moved out of the loop for performance...